zkevm-circuits/src/mpt_circuit/branch/extension_node.rs

```diff
@@ -154,1177 +190,138 @@ Key extension is [0].
-
-            let is_account_leaf_in_added_branch = meta.query_advice(
-                is_account_leaf_in_added_branch,
-                Rotation(rot_into_branch_init - 1),
-            );
-
-            // When placeholder extension, we don't check its hash in a parent.
-            let mut is_branch_placeholder = s_main.bytes[IS_BRANCH_S_PLACEHOLDER_POS - RLP_NUM];
-            if !is_s {
-                is_branch_placeholder = s_main.bytes[IS_BRANCH_C_PLACEHOLDER_POS - RLP_NUM];
-            }
-            let is_branch_placeholder =
-                meta.query_advice(is_branch_placeholder, Rotation(rot_into_branch_init));
-
-            let mut is_ext_node_non_hashed = s_main.bytes[IS_S_EXT_NODE_NON_HASHED_POS - RLP_NUM];
-            if !is_s {
-                is_ext_node_non_hashed = s_main.bytes[IS_C_EXT_NODE_NON_HASHED_POS - RLP_NUM];
+                ifx! {ext.is_longer_than_55_s => {
+                    require!(a!(s_main.rlp1) => RLP_LIST_LONG + 1);
+                }}
+                // Verify that the lenghts are consistent.
+                require!(ext.ext_len(meta, 0) => ext.ext_key_num_bytes(meta, 0) + ext.ext_branch_num_bytes(meta));
             }
-            let is_ext_node_non_hashed =
-                meta.query_advice(is_ext_node_non_hashed, Rotation(rot_into_branch_init));
 
-            let acc_c = meta.query_advice(accs.acc_c.rlc, Rotation::cur());
-
-            // Any rotation that lands into branch can be used instead of -21.
-            let mut mod_node_hash_rlc_cur = meta.query_advice(accs.s_mod_node_rlc, Rotation(-21));
-            if !is_s {
-                mod_node_hash_rlc_cur = meta.query_advice(accs.c_mod_node_rlc, Rotation(-21));
-            }
-
-            let selector = not_first_level
-                * q_enable
-                * (one.clone() - is_account_leaf_in_added_branch)
-                * (one.clone() - is_branch_placeholder)
-                * (one.clone() - is_ext_node_non_hashed);
-
-            let mut table_map = Vec::new();
-            let keccak_is_enabled = meta.query_advice(keccak_table.is_enabled, Rotation::cur());
-            table_map.push((selector.clone(), keccak_is_enabled));
-
-            let keccak_input_rlc = meta.query_advice(keccak_table.input_rlc, Rotation::cur());
-            table_map.push((selector.clone() * acc_c, keccak_input_rlc));
-
-            let mut rot = 0;
-            if !is_s {
-                rot = -1;
+            // Calculate the extension node RLC.
+            // The intermediate RLC after `s_main` bytes needs to be properly computed.
+            // s_rlp1, s_rlp2, s_bytes need to be the same in both extension rows.
+            // However, to make space for nibble witnesses, we put nibbles in
+            // extension row C s_bytes. So we use s_bytes from S row.
+            // TODO(Brecht): Do we need to store the RLC here? we can just use `rlc`
+            // directly below...
```

Comment:
Yes, I think you are right. We could compute the value that is now in `ext_rlc.prev` (no need to store it), but we need to store the value that is in `ext_rlc.cur` because it's needed in `require_in_parent`.